docs/manual/src/udl/custom_types.md

```diff
@@ -1,72 +1,105 @@
 # Custom types
 
-Custom types allow you to extend the UniFFI type system to support types from your Rust crate or 3rd
-party libraries. This works by converting to and from some other UniFFI type to move data across the
-FFI. You must provide a `UniffiCustomTypeConverter` implementation to convert the types.
+Custom types offer a way to use user-defined types in your interface that don't derive one of the UniFFI type traits (`uniffi::Record`, `uniffi::Enum`, `uniffi::Object`, etc.).
+Instead, custom types are converted to/from a existing type that does derive those traits when being passed across the FFI.
+
+Custom types are often used with structs that wraps a primitive type, for example `Guid(String)`.
+These types can be passed across the FFI as the primitive type, which can be more efficient than passing a struct.
+The foreign bindings will treat these types as the primitive type, for example `Guid` could appear a string to the foreign code.
+
+Custom types can also be customized on the foreign side.  For example, a URL could be:
+* Represented by the `url::Url` type in Rust
+* Passed across the FFI as a string
+* Represented by the `java.net.URL` type in Kotlin
 
 ## Custom types in the scaffolding code
 
-Consider the following trivial Rust abstraction for a "handle" which wraps an integer:
+### custom_type!
+
+Use the `custom_type!` macro to define a new custom type.
 
 ```rust
-pub struct Handle(i64);
+
+// Some complex struct that can be serialized/deserialized to a string.
+use some_mod::SerializableStruct;
+
+/// `SerializableStruct` objects will be passed across the FFI the same way `String` values are.
+uniffi::custom_type!(SerializableStruct, String);
 ```
 
-In this trivial example, the simplest way to expose this is with a macro.
+By default:
+
+ - Values passed to the foreign code will be converted using `<SerializableStruct as TryInto<String>>` then lowered as a `String`.
+     - The `TryInto::Error` type can be anything that implements `Into<anyhow::Error>`.
+     - If `String` implements `TryFrom<SerializableStruct>` this will also work, since there's a blanket impl in the core library. 
+ - Values passed to the Rust code will lifted as a `String` then converted using `<String as Into<SerializableStruct>>`.
+     - If `SerializableStruct` implements `From<String>` this will also work, since there's a blanket impl in the core library. 
+
+### custom_type! with manual conversions
 
+You can also manually specify the conversions by passing an extra param to the macro:
+
+```rust
+uniffi::custom_type!(SerializableStruct, String, {
+    into_existing: |s| s.serialize(),
+    try_from_existing: |s| s.deserialize(),
+});
 ```
+
+### custom_newtype!
+
+The `custom_newtype!` can trivially handle newtypes that wrap a UniFFI type.
```

Comment:
I would not put "trivially" in any sort of documentation. Maybe:

> The `custom_newtype!` macro is able to handle custom Rust structs which wrap a UniFFI type by default.

I don't want to go there yet, but the `custom_type!` macro name could be a bit more telling. It's not at all obvious what it does.

Reply:
Good point.  I adjusted the language slightly, but hopefully captured your overall point here.  Feel free to re-open if there are more changes here though.